cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/visitor/r4/KnowledgeArtifactPackageVisitor.java

```diff
@@ -187,4 +221,368 @@ public static List<BundleEntryComponent> findUnsupportedInclude(
+            }
+        }
+
+        return resourceList;
+    }
+
+    /**
+     * search by versioned Canonical URL
+     * @param url canonical URL of the form www.example.com/Patient/123|0.1
+     * @param repository to do the searching
+     * @return a bundle of results
+     */
+    private Bundle searchResourceByUrl(String url, Repository repository) {
+        Map<String, List<IQueryParameterType>> searchParams = new HashMap<>();
+
+        List<IQueryParameterType> urlList = new ArrayList<>();
+        urlList.add(new UriParam(Canonicals.getUrl(url)));
+        searchParams.put("url", urlList);
+
+        List<IQueryParameterType> versionList = new ArrayList<>();
+        String version = Canonicals.getVersion(url);
+        if (version != null && !version.isEmpty()) {
+            versionList.add(new TokenParam(Canonicals.getVersion((url))));
+            searchParams.put("version", versionList);
+        }
+
+        Bundle searchResultsBundle = (Bundle) repository.search(
+                Bundle.class, ResourceClassMapHelper.getClass(Canonicals.getResourceType(url)), searchParams);
+        return searchResultsBundle;
+    }
+
+    /**
+     * Removes any existing UsageContexts corresponding to the VSM specific extensions
+     * @param usageContexts the list of usage contexts to modify
+     */
+    private List<UsageContext> removeExistingReferenceExtensionData(List<UsageContext> usageContexts) {
+        List<String> useContextCodesToReplace = Collections.unmodifiableList(Arrays.asList(
+                KnowledgeArtifactAdapter.valueSetConditionCode, KnowledgeArtifactAdapter.valueSetPriorityCode));
+        return usageContexts.stream()
+                // remove any useContexts which need to be replaced
+                .filter(useContext -> !useContextCodesToReplace.stream()
+                        .anyMatch(code -> useContext.getCode().getCode().equals(code)))
+                .collect(Collectors.toList());
+    }
+
+    private void tryAddCondition(List<UsageContext> usageContexts, CodeableConcept condition) {
+        boolean focusAlreadyExists = usageContexts.stream()
+                .anyMatch(u -> u.getCode().getSystem().equals(KnowledgeArtifactAdapter.contextTypeUrl)
+                        && u.getCode().getCode().equals(KnowledgeArtifactAdapter.valueSetConditionCode)
+                        && u.getValueCodeableConcept()
+                                .hasCoding(
+                                        condition.getCoding().get(0).getSystem(),
+                                        condition.getCoding().get(0).getCode()));
+        if (!focusAlreadyExists) {
+            UsageContext newFocus = new UsageContext(
+                    new Coding(
+                            KnowledgeArtifactAdapter.contextUrl, KnowledgeArtifactAdapter.valueSetConditionCode, null),
+                    condition);
+            newFocus.setValue(condition);
+            usageContexts.add(newFocus);
+        }
+    }
```

Comment:
Nitpick: lines between methods.
```suggestion
    }
    
```